src/CRFTDRegistry.sol

```diff
@@ -27,16 +27,42 @@ error ImplementationNotApproved();
 /// @notice CRFTD proxy registry
 contract CRFTDRegistry is Owned(msg.sender) {
     event Registered(address indexed user, bytes32 id, uint256 fee);
+    event CollectionRegistered(address indexed user, bytes32 id, uint256 fee);
     event ProxyDeployed(address indexed owner, address indexed proxy);
+    event VIPStatusChanged(address indexed user, bool status);
 
     mapping(address => bool) public approvedImplementation;
 
-    /* ------------- external ------------- */
+    mapping(bytes32 => uint256) public paidStatus;
+    mapping(address => bool) public vipRole;
+
 
+    /* ------------- external ------------- */
     function register(bytes32 id) external payable {
         emit Registered(msg.sender, id, msg.value);
     }
 
+    /// @dev This is reponsible for the collect payment for art generation
+    /// @dev tokenId is the keccak256(abi.encodePacked(msg.sender,collectionId,collectionSize))
+    function registerCollection(bytes32 tokenId, uint256 collectionSize) external payable {
```

Comment:
Should we also have an additional check in this `register` function to check whether the passed in `tokenSetId` has already been paid for?

Something like..
`if (paidExports[tokenSetId]) revert AlreadyPaid();`

Reply:
I don't think we really need this because it's public mapping so we can add this check on frontend but let me know your further thought for it.